qualtran/bloqs/rotations/phase_gradient.py

```diff
@@ -150,27 +152,39 @@ class AddIntoPhaseGrad(GateWithRegisters, cirq.ArithmeticGate):
         [Compilation of Fault-Tolerant Quantum Heuristics for Combinatorial Optimization](https://arxiv.org/abs/2007.07391)
         Appendix A: Addition for controlled rotations
     """
+
     x_bitsize: int
     phase_bitsize: int
     right_shift: int = 0
     sign: int = +1
+    controlled: Optional[bool] = None
```

Comment:
Please update the docstring to include the `controlled` arg in the `Args` and `Registers` section. Also change to `controlled: Optional[int] = None` s.t. `controlled=0` means control should be active when ctrl register stores value `0`. I think right now, your implementation below works only for the case when control register stores value `1` (i.e. controlled=True)